src/OpenSage.Game/Audio/AudioBuffer.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using OpenSage.Content;
+using OpenSage.Data.Wav;
+using OpenAL;
+
+namespace OpenSage.Audio
+{
+    public sealed class AudioBuffer : DisposableBase
+    {
+        private uint _handle;
+        private List<AudioSource> _usageList;
+
+        public List<AudioSource> UsageList => _usageList;
+        public uint Handle => _handle;
+
+        public AudioBuffer()
+        {
+            AL10.alGenBuffers(1, out _handle);
+            AudioSystem.alCheckError();
+            _usageList = new List<AudioSource>();
+        }
+
+        public AudioBuffer(ContentManager contentManager, WavFile wavFile) : this()
+        {
+            int format = 0;
+
+            switch (wavFile.Channels)
+            {
+                case 1:
+                    {
+                        switch (wavFile.BitsPerSample)
+                        {
+                            case 8:
+                                format = AL10.AL_FORMAT_MONO8;
+                                break;
+                            case 16:
+                                format = AL10.AL_FORMAT_MONO16;
+                                break;
+                            default:
+                                throw new NotSupportedException("Invalid audio format!");
+                        }
+                    }
+                    break;
+                case 2:
+                    {
+                        switch (wavFile.BitsPerSample)
+                        {
+                            case 8:
+                                format = AL10.AL_FORMAT_STEREO8;
+                                break;
+                            case 16:
+                                format = AL10.AL_FORMAT_STEREO16;
+                                break;
+                            default:
+                                throw new NotSupportedException("Invalid audio format!");
+                        }
+                    }
+                    break;
+            }
+
+
+            AL10.alBufferData(_handle, format, wavFile.Buffer, wavFile.Size, wavFile.Fequency);
+            AudioSystem.alCheckError();
+        }
+
+        public TimeSpan Duration
```

Comment:
You mentioned this can't be calculated upfront in the constructor because you want to re-use this class for streamed audio. If so, I think it would be better to make this a `GetDuration()` method. .NET guidelines suggest using methods instead of properties for "expensive" computations like this one.